core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/WorkflowResource.scala

```diff
@@ -104,6 +104,32 @@ class WorkflowResource {
     }
   }
 
+  /**
+    * This method creates and insert a new workflow to database
+    *
+    * @param session  HttpSession
+    * @param workflow , a workflow
+    * @return Workflow, which contains the generated wid if not provided
+    */
+  @POST
+  @Path("/create")
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  @Produces(Array(MediaType.APPLICATION_JSON))
+  def createWorkflow(@Session session: HttpSession, workflow: Workflow): Response = {
+    UserResource.getUser(session) match {
+      case Some(user) =>
+        if (workflowOfUserExists(workflow.getWid, user.getUid)) {
+          Response.status(Response.Status.BAD_REQUEST).build()
+        } else {
+          workflowDao.insert(workflow)
+          workflowOfUserDao.insert(new WorkflowOfUser(user.getUid, workflow.getWid))
+          Response.ok(workflowDao.fetchOneByWid(workflow.getWid)).build()
```

Comment:
workflow is already fulfilled with wid, no need to fetch again.

Reply:
I actually don't quite understand your meaning, if I change the code in line 126 to 
`Response.ok(workflow).build()`
In the frontend, the feedback workflow it gets won't have the creation time and modified time, so for the a newly duplicated workflow , time won't appear on the dashboard 
It seems that the two time field won't be fulfilled as the wid do, so I think we still need to fetch it.
But I am also confused why this happens.

Reply:
ok we can fetch it again.